v3/src/components/web-view/use-data-interactive-controller.tsx

```diff
@@ -0,0 +1,74 @@
+import { useToast } from "@chakra-ui/react"
+import iframePhone from "iframe-phone"
+import React, { useEffect } from "react"
+import { DEBUG_PLUGINS, debugLog } from "../../lib/debug"
+import { ITileModel } from "../../models/tiles/tile-model"
+import { isWebViewModel } from "./web-view-model"
+
+function extractOrigin(url?: string) {
+  if (!url) return
+  return new URL(url).origin
```

Comment:
I realize this was my suggestion (and I stick by it), but it occurred to me after I made it that it's worth checking what the behavior of `URL` is when given an invalid url string. May need to wrap this in `try`/`catch` or something like that. If the answer is non-obvious then perhaps a TODO would suffice for now.